docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+To verify the control plane is up:
+
+```bash
+kubectl get kubeadmcontrolplane
+```
+
+You should see an output similar to the below:
+
+```bash
+NAME                            INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
+clusterclass-quickstart         true                                 v1.21.2   1                  1         1
+```
+
+
+### Upgrade a Cluster using Managed Topology
+
+The Spec.Topology field added to the Cluster object as part of ClusterClass allows changes made on the Cluster to be propogated across all relevant objects. This turns a Kubernetes cluster upgrade into a one touch operation.
+Looking at the cluster we just created we can see the version of the control plane and the other machines is v1.21.2.
+
+```bash
+> kubectl get kubeadmcontrolplane,machines
+
+NAME                            INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
+clusterclass-quickstart         true          true                   v1.21.2   1                  1         1
+
+
+NAME                                                           CLUSTER                   AGE     PROVIDERID                                                                PHASE     VERSION
+clusterclass-quickstart-linux-workers-XXXX                     clusterclass-quickstart   2m51s   docker:////clusterclass-quickstart-linux-workers-XXXX                    Running   v1.21.2
+clusterclass-quickstart-XXXX                                   clusterclass-quickstart   2m48s   docker:////clusterclass-quickstart-XXXX                                  Running   v1.21.2
+
+```
+
+To update the Cluster the only change needed is to the 'version' field under spec.topology in the Cluster object. Change '1.21.2' to '1.22' as below. 
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: Cluster
+metadata:
+  name: "clusterclass-quickstart"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: cluster-class
+    version: v1.22.0
+      controlPlane:
+      metadata:
+        labels:
+          topologyLabel: topologyLabelValue
+        annotations:
+          topologyAnnotation: topologyAnnotation
+      replicas: 1 
+    workers:
+      machineDeployments:
+        - class: linux-worker
+          name: linux-workers
+          metadata:
+            labels:
+              topologyLabel: topologyLabelValue
+            annotations:
+              topologyAnnotation: topologyAnnotation
+          replicas: 1 
+```
+
+Save the above change to your local file and apply the update to the Cluster with:
+```bash
+kubectl apply -f clusterclass.yaml
+```
+
+The upgrade will take a minute to roll out as it will take place machine by machine with older versions of the machines only being removed after health newer versions come online.
```

Comment:
```suggestion
The upgrade will take a bit to roll out as it will take place machine by machine with older versions of the machines only being removed after healthy newer versions come online.
```
(not sure about "a minute" :))